subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java

```diff
@@ -70,7 +73,10 @@
      * @param main the fully qualified name of the main class to be executed.
      *
      * @return this
+     *
+     * @deprecated Use {@link #getMainClass()}.set(main) instead. This method will be removed in Gradle 8.0.
      */
+    @Deprecated
```

Comment:
Please add the `@ReplacedBy` annotation here too.

Reply:
Done.

Reply:
I had to remove this again as there is some validation in the pipeline that breaks the build when `@ReplacedBy` is put on a setter which tells to only add this annotation to getters.